### PR TITLE
fix(web_driver_docker_client): don't fail on BadHostKeyException

### PR DESCRIPTION
**Pull request by dkropachev**
https://trello.com/c/6IrK5eh4/2725-address-badhostkeyexception-issue

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/utils/remotewebbrowser.py [hunk 1]
```diff
@@ -46,7 +47,15 @@ def web_driver_docker_client(self) -> Optional[DockerClient]:
         SSHAgent.add_keys((self.ssh_login_info["key_file"], ))
         # since a bug in docker package https://github.com/docker-library/python/issues/517 that need to explicitly
         # pass down the port for supporting ipv6
-        return DockerClient(base_url=f"ssh://{self.ssh_login_info['user']}@{normalize_ipv6_url(self.ssh_login_info['hostname'])}:22",
+        try:
+            return DockerClient(base_url=f"ssh://{self.ssh_login_info['user']}@{normalize_ipv6_url(self.ssh_login_info['hostname'])}:22",
+                            timeout=DOCKER_API_CALL_TIMEOUT)
+        except paramiko.ssh_exception.BadHostKeyException as exc:
+            system_host_key_path = os.path.expanduser("~/.ssh/known_hosts")
```
Comment (enaydanov) [inline sdcm/utils/remotewebbrowser.py:54]: small cosmetic issue: s/key/keys/

**Review by enaydanov — APPROVED** [review]

File: sdcm/utils/remotewebbrowser.py [hunk 2]
```diff
@@ -13,17 +13,18 @@
...
 
+import paramiko
 from docker import DockerClient  # pylint: disable=wrong-import-order
 from selenium.webdriver import Remote, ChromeOptions
 
 from sdcm.utils.docker_utils import ContainerManager, DOCKER_API_CALL_TIMEOUT
 from sdcm.utils.common import get_free_port, wait_for_port, normalize_ipv6_url
 from sdcm.utils.ssh_agent import SSHAgent
 
-
```
Comment (enaydanov) [inline sdcm/utils/remotewebbrowser.py:26]: Please, return this empty line back :)

File: sdcm/utils/remotewebbrowser.py [hunk 3]
```diff
@@ -46,7 +47,15 @@ def web_driver_docker_client(self) -> Optional[DockerClient]:
         SSHAgent.add_keys((self.ssh_login_info["key_file"], ))
         # since a bug in docker package https://github.com/docker-library/python/issues/517 that need to explicitly
         # pass down the port for supporting ipv6
-        return DockerClient(base_url=f"ssh://{self.ssh_login_info['user']}@{normalize_ipv6_url(self.ssh_login_info['hostname'])}:22",
+        try:
+            return DockerClient(base_url=f"ssh://{self.ssh_login_info['user']}@{normalize_ipv6_url(self.ssh_login_info['hostname'])}:22",
+                            timeout=DOCKER_API_CALL_TIMEOUT)
```
Comment (enaydanov) [inline sdcm/utils/remotewebbrowser.py:52]: Pre-commit is not happy with this indentation.

File: sdcm/utils/remotewebbrowser.py [hunk 4]
```diff
@@ -46,7 +47,15 @@ def web_driver_docker_client(self) -> Optional[DockerClient]:
...
+        except paramiko.ssh_exception.BadHostKeyException as exc:
+            system_host_key_path = os.path.expanduser("~/.ssh/known_hosts")
+            system_host_keys = paramiko.hostkeys.HostKeys(system_host_key_path)
+            if system_host_keys.pop(exc.hostname, None):
+                system_host_keys.save(system_host_key_path)
+            return DockerClient(base_url=f"ssh://{self.ssh_login_info['user']}@{normalize_ipv6_url(self.ssh_login_info['hostname'])}:22",
                             timeout=DOCKER_API_CALL_TIMEOUT)
```
Comment (enaydanov) [inline sdcm/utils/remotewebbrowser.py:59]: ...and here too.

File: sdcm/utils/remotewebbrowser.py (hunk 3 above)
Comment (dkropachev) [reply sdcm/utils/remotewebbrowser.py]: fixed

File: sdcm/utils/remotewebbrowser.py (hunk 1 above)
Comment (dkropachev) [reply sdcm/utils/remotewebbrowser.py]: fixed

File: sdcm/utils/remotewebbrowser.py (hunk 4 above)
Comment (dkropachev) [reply sdcm/utils/remotewebbrowser.py]: fixed

File: sdcm/utils/remotewebbrowser.py (hunk 2 above)
Comment (dkropachev) [reply sdcm/utils/remotewebbrowser.py]: done

**Review by enaydanov — APPROVED** [review]

**Pull request merged**